### PR TITLE
API - Unit Test - Init Tests For Search User And Get User Info

### PR DESCRIPTION
**Pull request by at-haole2**
#### Related issues (Required)
- Task #226 
- Bug #

#### Description (Required)
-  Init Tests For Search User And Get User Info

### CODE REVIEW

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">All test passed. (21/21)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        /* Then */
+        val request = server.takeRequest()
+        Assert.assertThat(request.method.toUpperCase(), `is`("GET"))
+        Assert.assertThat(request.requestUrl.pathSegments(), hasItems("users", "{userId}"))
+
+        test.assertValue { it.id == "a7f13570-5e5e-4651-a58d-985bfcfcedf7" }
+        test.assertValue { it.groupId == "5bb1367e-8941-4874-8897-0bb84f154bfa" }
+        test.assertValue { it.name == "rim" }
+    }
+}
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt]: add break line

File: app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        /* When */
+        restClient.getUserInfo("a7f13570-5e5e-4651-a58d-985bfcfcedf7").subscribe(test)
+
+        /* Then */
+        val request = server.takeRequest()
+        Assert.assertThat(request.method.toUpperCase(), `is`("GET"))
+        Assert.assertThat(request.requestUrl.pathSegments(), hasItems("users", "{userId}"))
+
+        test.assertValue { it.id == "a7f13570-5e5e-4651-a58d-985bfcfcedf7" }
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt]: change this 3 line to 
```
test.assertValue {
            assertThat(it.id, `is`("a7f13570-5e5e-4651-a58d-985bfcfcedf7"))
            assertThat(it.groupId, `is`("5bb1367e-8941-4874-8897-0bb84f154bfa"))
            assertThat(it.name, `is`("rim"))
            true
        }
```

File: app/src/test/resources/responses/getUserInfo.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "sub_status_text": "Last updated 15 minutes ago",
+    "seconds_elapsed_since_last_heartbeat": 950.702721,
+    "battery": 69,
+    "status_text": "Location disabled",
+    "speed": null,
+    "is_warning": true
+  },
+  "created_at": "2017-12-07T04:42:55.635339Z",
+  "modified_at": "2017-12-18T06:46:21.637335Z"
+}
```
Comment (at-vinhhuynh) [inline app/src/test/resources/responses/getUserInfo.json]: add break line

File: app/src/test/resources/responses/searchUser.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        "battery": 77,
+        "status_text": "Location disabled",
+        "speed": null,
+        "is_warning": true
+      },
+      "created_at": "2017-12-07T04:42:55.635339Z",
+      "modified_at": "2017-12-18T02:37:12.642250Z"
+    }
+  ]
+}
```
Comment (at-vinhhuynh) [inline app/src/test/resources/responses/searchUser.json]: add break line

File: app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Test
+    fun `Given mock response - When request getUserInfo - Then return user object`() {
+        /* Given */
+        val test = TestObserver<User>()
+        server.addResponseBody("getUserInfo.json")
+
+        /* When */
+        restClient.getUserInfo("a7f13570-5e5e-4651-a58d-985bfcfcedf7").subscribe(test)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt]: create a variable for this value `a7f13570-5e5e-4651-a58d-985bfcfcedf7`

File: app/src/test/java/vn/asiantech/way/api/SearchUserTest.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Test
+    fun `Given mock response - When request searchUser - Then return UserListResult object`() {
+        /* Given */
+        val test = TestObserver<UserListResult>()
+        server.addResponseBody("searchUser.json")
+
+        /* When */
+        restClient.searchUser("rim").subscribe(test)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/SearchUserTest.kt]: please define a variable for this value `rim `

**Comment by at-haole2** [comment]
I fixed comments. Please check again. Thanks!

File: app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        /* When */
+        restClient.getUserInfo(userId).subscribe(test)
+
+        /* Then */
+        val request = server.takeRequest()
+        Assert.assertThat(request.method.toUpperCase(), `is`("GET"))
+        Assert.assertThat(request.requestUrl.pathSegments(), hasItems("users", "{userId}"))
+
+        test.assertValue {
+            assertThat(it.id, `is`("a7f13570-5e5e-4651-a58d-985bfcfcedf7"))
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GetUserInfoTest.kt]: please use `userId ` instead for `a7f13570-5e5e-4651-a58d-985bfcfcedf7`

File: app/src/test/java/vn/asiantech/way/api/SearchUserTest.kt [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        Assert.assertThat(request.method.toUpperCase(), `is`("GET"))
+        Assert.assertThat(request.requestUrl.queryParameterNames(), hasItems("name"))
+        Assert.assertThat(request.requestUrl.queryParameter("name"), `is`("rim"))
+
+        test.assertValue {
+            val item = it.users[0]
+            Assert.assertThat(it.users.size, `is`(1))
+            Assert.assertThat(item.id, `is`("a7f13570-5e5e-4651-a58d-985bfcfcedf7"))
+            Assert.assertThat(item.name, `is`("rim"))
+            it.users.isNotEmpty()
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/SearchUserTest.kt]: please use `name` instead for `rim`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**